conda_smithy/schema/models.py

```diff
@@ -0,0 +1,476 @@
+    ] = False
+
+    check_solvable: Annotated[
+        Union[bool, None],
+        Field(
+            description="Open PRs only if resulting environment is solvable."
+        ),
+    ] = None
+
+    inspection: Annotated[
+        Union[BotConfigInspectionChoice, None],
+        Field(
+            description="Method for generating hints or updating recipe",
+        ),
+    ] = None
+
+    abi_migration_branches: Annotated[
+        Union[List[str], None],
+        Field(description="List of branches for additional bot migration PRs"),
+    ] = None
+
+    version_updates_random_fraction_to_keep: Annotated[
+        Union[float, None],
+        Field(
+            description="Fraction of versions to keep for frequently updated packages"
+        ),
+    ] = None
+
+
+class ChannelPriorityConfig(str, Enum):
+    STRICT = "strict"
+    FLEXIBLE = "flexible"
+    DISABLED = "disabled"
+
+
+class CondaForgeChannels(BaseModel):
+    """This represents the channels to grab packages from during builds and which channels/labels to push to on anaconda.org after a package has been built. The channels variable is a mapping with sources and targets."""
+
+    sources: List[str] = Field(
+        default=["conda-forge"],
+        description="sources selects the channels to pull packages from, in order",
+    )
+
+    targets: List[List[str]] = Field(
+        default=[["conda-forge", "main"]],
+        description="targets is a list of 2-lists, where the first element is the channel to push to and the second element is the label on that channel",
+    )
+
+
+class CondaBuildConfig(BaseModel):
+    pkg_format: Union[int, str, None] = Field(
+        description="The package format for conda build. This can be one of 1, 2, or tar. The default is 2.",
+        default="2",
+    )
+
+    zstd_compression_level: int = Field(
+        default=16,
+        description="The compression level for the zstd compression algorithm for .conda artifacts. conda-forge uses a default value of 16 since its artifacts can be large.",
+    )
+
+    @field_validator("pkg_format")
+    def pkg_format_validator(cls, v):
+        if not v:
+            return v
+
+        valid_values = ["2", None]
+        if v not in valid_values:
+            raise ValueError(
+                f"Invalid value {v} for pkg_format. Valid values are {valid_values}"
+            )
+        return v
+
+
+class CondaForgeDocker(BaseModel):
+    model_config = ConfigDict(extra="allow")
+
+    executable: str = Field(
+        description="The executable for Docker", default="docker"
+    )
+
+    fallback_image: str = Field(
+        description="The fallback image for Docker",
+        default="quay.io/condaforge/linux-anvil-comp7",
+    )
+
+    command: str = Field(
+        description="The command to run in Docker", default="bash"
+    )
+
+    interactive: bool = Field(
+        description="Whether to run Docker in interactive mode", default=None
```

Comment:
The default should be a bool; I guess `false`?

Reply:
The reason for it being none is that under the original method dictionary https://github.com/conda-forge/conda-smithy/blob/main/conda_smithy/configure_feedstock.py#L1716, those fields do not exist, and to have the same behavior you need to set those as none and under the dump method make sure its set to not render none values.

In any case those are quick fixes in both sides. tks